### PR TITLE
Implem cv

### PR DESCRIPTION
**Pull request by CedricAllain**
Modify objective.py to add the cross-validation split.
Comments about how cv works in bechopt is added in the get_objective() method of the objective.

Create a solver template, to demonstrate that it needs to return a ForecastModel instance with a predict() method.

### CODE REVIEW

File: datasets/simulated.py [hunk 1]
```diff
@@ -20,10 +20,11 @@ class Dataset(BaseDataset):
     # Any parameters 'param' defined here is available as `self.param`.
     parameters = {
         "n_features": [5],
-        "n_windows": [10],
-        "window_size": [512],
-        "horizon": [96],
-        "random_state": [42],
+        "n_samples": [1_000],
+        # "n_windows": [10],
```
Comment (Zaaaazou) [inline datasets/simulated.py:24]: to be deleted

File: datasets/simulated.py [hunk 2]
```diff
@@ -38,19 +39,22 @@ def get_data(self):
...
+        # X_test = rng.randn(n_windows_test, self.n_features, self.window_size)
+        # y_test = rng.randn(n_windows_test, self.n_features, self.horizon)
 
-        X = check_data(X)
-        y = check_data(y)
+        # X = (X_train, X_test)
+        # y = (y_train, y_test)
+
+        # X = check_data(X)
+        # y = check_data(y)
```
Comment (Zaaaazou) [inline datasets/simulated.py:57]: to be deleted

**Review by Zaaaazou — CHANGES_REQUESTED** [review]

File: datasets/simulated.py (hunk 1 above)
Comment (Zaaaazou) [inline datasets/simulated.py:24]:
```

File: datasets/simulated.py [hunk 3]
```diff
@@ -20,10 +20,11 @@ class Dataset(BaseDataset):
...
         "n_features": [5],
-        "n_windows": [10],
-        "window_size": [512],
-        "horizon": [96],
-        "random_state": [42],
+        "n_samples": [1_000],
+        # "n_windows": [10],
+        # "window_size": [512],
+        # "horizon": [96],
+        # "random_state": [42],
```
Comment (Zaaaazou) [inline datasets/simulated.py:27]:
```suggestion
        # "window_size": [512],
        # "horizon": [96],
        # "random_state": [42],
```
```suggestion

```

**Review by Zaaaazou — CHANGES_REQUESTED** [review]

File: objective.py [hunk 4]
```diff
@@ -18,14 +19,30 @@ class Objective(BaseObjective):
...
     # List of packages needed to run the benchmark.
     # They are installed with conda; to use pip, use 'pip:packagename'. To
     # install from a specific conda channel, use 'channelname:packagename'.
     # Packages that are not necessary to the whole benchmark but only to some
     # solvers or datasets should be declared in Dataset or Solver (see
     # simulated.py and python-gd.py).
     # Example syntax: requirements = ['numpy', 'pip:jax', 'pytorch:pytorch']
-    requirements = []
+    install_cmd = "conda"
+    requirements = [numpy, sklearn]
```
Comment (Jad-yehya) [inline objective.py:45]: bad requirements specification:

`requirements = ["scikit-learn"]`

No need to specify numpy

File: solvers/solver_template.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+class Solver(BaseSolver):
+
+    # Name to select the solver in the CLI and to display the results.
+    name = "solver_template"
+
+    # List of parameters for the solver. The benchmark will consider
+    # the cross product for each key in the dictionary.
+    # All parameters 'p' defined here are available as 'self.p'.
+    parameters = {
+        "param1": [1],
```
Comment (Jad-yehya) [inline solvers/solver_template.py:23]: What is params1 ?

**Review by Jad-yehya — CHANGES_REQUESTED** [review]

File: objective.py (hunk 4 above)
Comment (Jad-yehya) [reply objective.py]: Fixed

**Pull request merged**